pkg/kubelet/status/status_manager.go

```diff
@@ -705,6 +705,11 @@ func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUp
 	// Set DisruptionTarget.LastTransitionTime.
 	updateLastTransitionTime(&status, &oldStatus, v1.DisruptionTarget)
 
+	// Set the observedGeneration for this pod status. Static pods are excluded.
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodObservedGenerationTracking) && !metav1.HasAnnotation(pod.ObjectMeta, v1.MirrorPodAnnotationKey) {{
+		status.ObservedGeneration = pod.GetGeneration()
+	}
```

Comment:
`updateStatusInternal` is also called when updating probe states, which I don't think we want bump the observed generation. I think it's better to move this to the `SetPodStatus` method.

Reply:
I think you should also set the status in the readiness conditions (line 400), or were you planning to tackle conditions in a separate PR?

Reply:
What was decided in the KEP https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/5067-pod-generation#beta is that implementation of conditions is not required until beta. 

I'm not sure if that means I _shouldn't_ add them in alpha, or just that it won't be blocking for alpha. If it's the latter I'm happy to add this here now.

Reply:
I tried to look through the node lifecycle controller and scheduler code to see where they update the status, and AFAICT most of the time where they are updating the status, they are setting pod conditions (so they don't need to update the top-level `status.observedGeneration`, they'll just need to set the observedGeneration on the condition). So I added what I _think_ is all that is needed for the scheduler for setting status.observedGeneration to this PR, but please feel free to double check me